src/typegraph/deno/src/io.ts

```diff
@@ -7,6 +7,38 @@ import process from "node:process";
  * see: module level documentation `meta-cli/src/deploy/actors/task.rs`
  */
 
+const JSONRPC_VERSION = "2.0";
+
+function writeRpcMessage(message: string) {
+  // split into 32-KiB chunks
+  const chunkSize = 32758; // 32 KiB - 10 bytes for "jsonrpc^: " or "jsonrpc$: "
+  for (let i = 0; i < message.length; i += chunkSize) {
+    const chunk = message.slice(i, i + chunkSize);
```

Comment:
Since the message length returns the number of code units not the characters. With string slice there is still a risk of cutting in the middle of a character.
I suggest using `Array.from(message)` to get an actual iterator to the characters then perform slice on that. That would also mean additional fixes around the actual total length though.. but if it's about sending raw bytes it should be fine I guess.

```js
const text = "Hello👍";
console.log(text.length); // 7 not 6
console.log(text.slice(0, 6));
console.log(Array.from(text).slice(0, 6). join(""));
```

Reply:
Good point... Each entry is parsed as a string in the rust size, so UTF validation might fail...
Unless the implementation follows this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#well-formed_json.stringify